portal-impl/test/unit/com/liferay/portal/zip/ZipWriterImplTest.java

```diff
@@ -305,6 +305,29 @@ public void testConstructor() {
 		file2.delete();
 	}
 
+	@Test
+	public void testConstructorWithFilePathContainSpecialCharacter()
+		throws IOException {
+
+		Path tempZipFilePath = Files.createTempFile(
+			Paths.get(SystemProperties.get(SystemProperties.TMP_DIR)), null,
+			"file path with space.zip");
+
+		Files.delete(tempZipFilePath);
+
+		File zipFile = new File(tempZipFilePath.toString());
+
+		ZipWriter zipWriter = new ZipWriterImpl(zipFile);
+
+		File file = zipWriter.getFile();
+
+		Assert.assertNotNull(file);
+		Assert.assertTrue(file.exists());
+		Assert.assertEquals(zipFile.getPath(), file.getPath());
+
+		file.delete();
```

Comment:
@JanisZhang I can see that you borrow the logic from other test cases in the class, but there's an issue with this approach. If any of the assert* throws an `AssertionError` then the file won't be deleted. So the file.delete() should either be put into a `tearDown()` or in the finally block in a try-finally. For now, please at least correct this one.

Reply:
@dantewang I got it, thanks!